### PR TITLE
Tie break top-scoring bandit actions by action name

### PR DESCRIPTION
**Pull request by aarsilv**
_Eppo Internal_
🎟️ **Ticket:** [FF-3004 - Tie break top scoring bandit actions by action name](https://linear.app/eppo/issue/FF-3004/js-common-sdk-tie-break-top-scoring-bandit-actions-by-action-name)

## Motivation and Context
There are certain situations where actions can be tied for the top score. In these situations, we want to deterministically declare a winner, as its weight is computed differently than the others. Currently, in this SDK, the winner is based on the order in which the actions are passed to the bandit--which is not what we want.

This situation is recreated in a newly added test case for our shared test data ([PR link](https://github.com/Eppo-exp/sdk-test-data/pull/52))

## Description
When determining the top-scoring action, prioritize actions with alphabetically earlier names.

## How has this been tested?
Will be tested by the new shared test case data. In the meantime, I manually copied the test files to run the test.

### CODE REVIEW

File: src/bandit-evaluator.ts [hunk 1]
```diff
@@ -144,7 +144,12 @@ export class BanditEvaluator {
     let currTopScore: number | null = null;
     let currTopAction: string | null = null;
     actionScoreEntries.forEach(([actionKey, actionScore]) => {
-      if (currTopScore === null || actionScore > currTopScore) {
+      if (
+        currTopScore === null ||
+        currTopAction === null ||
```
Comment (aarsilv) [inline src/bandit-evaluator.ts:149]: this superfluous check is to appease TypeScript

File: src/bandit-evaluator.ts [hunk 2]
```diff
@@ -144,7 +144,12 @@ export class BanditEvaluator {
...
+        actionScore > currTopScore ||
+        (actionScore === currTopScore && actionKey < currTopAction)
```
Comment (aarsilv) [inline src/bandit-evaluator.ts:151]: This line is the fix

**Review by giorgiomartini0 — APPROVED** [review]

File: src/bandit-evaluator.ts (hunk 2 above)
Comment (typotter) [reply src/bandit-evaluator.ts:151]: mind tossing in a quick comment in the code for future archeologists ?

File: src/bandit-evaluator.ts (hunk 1 above)
Comment (typotter) [reply src/bandit-evaluator.ts:149]: 🙇‍♂️ bow to the wishes of the TS Language Server.

**Review by typotter — APPROVED** [review]
:heart: the universal test cases exercising this codepath. Consider adding a unit test ([example](https://github.com/Eppo-exp/php-sdk/pull/32/files))

**Pull request merged**